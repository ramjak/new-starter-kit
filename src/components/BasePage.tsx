import * as React from "react";
import { AppBar, Card, Button, Toolbar, Typography } from "@material-ui/core";
import { Link } from "react-router-dom";
import styles from "./BasePage.module.scss";
import logo from "../logo.svg";

export interface ILink {
  text: string;
  link: string;
}

interface IBasePage {
  children: React.ReactNode;
  topNavLinks: ILink[];
}

function BasePage(props: IBasePage) {
  const { children, topNavLinks } = props;
  return (
    <div className={styles.App}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" className={styles.appTitle}>
            Sample App
          </Typography>
          {topNavLinks.map((nav) => (
            <Button
              key={nav.link}
              component={Link}
              to={nav.link}
              className={styles.navButton}
            >
              {nav.text}
            </Button>
          ))}
        </Toolbar>
      </AppBar>
      <div className={styles.contentWrapper}>
        <img src={logo} className={styles.AppLogo} alt="logo" />
        <Card className={styles.WrapperCard}>{children}</Card>
      </div>
    </div>
  );
}

export default BasePage;
