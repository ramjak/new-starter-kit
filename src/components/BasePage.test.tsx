import "reflect-metadata";
import { createRenderer } from "react-test-renderer/shallow";
import BasePage from "./BasePage";

it("render BasePage", () => {
  const shallowRenderer = createRenderer();

  shallowRenderer.render(
    <BasePage
      topNavLinks={[
        { link: "/", text: "Root" },
        { link: "/test", text: "Test" },
      ]}
    >
      children
    </BasePage>
  );
  const result = shallowRenderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
