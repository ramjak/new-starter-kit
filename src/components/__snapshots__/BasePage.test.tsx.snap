// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render BasePage 1`] = `
<div
  className="App"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(Typography))
        className="appTitle"
        variant="h6"
      >
        Sample App
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/"
      >
        Root
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Button))
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        to="/test"
      >
        Test
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
  <header
    className="AppHeader"
  >
    <img
      alt="logo"
      className="AppLogo"
      src="logo.svg"
    />
    <WithStyles(ForwardRef(Card))
      className="WrapperCard"
    >
      children
    </WithStyles(ForwardRef(Card))>
  </header>
</div>
`;
